library/spdm_requester_lib/libspdm_req_key_update.c

```diff
@@ -268,29 +269,32 @@ static libspdm_return_t libspdm_try_key_update(libspdm_context_t *spdm_context,
         libspdm_release_receiver_buffer (spdm_context);
         return status;
     }
-    if (spdm_response_size < sizeof(spdm_message_header_t)) {
-        LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "SpdmVerifyKey[%x] Failed\n", session_id));
-        libspdm_release_receiver_buffer (spdm_context);
-        return LIBSPDM_STATUS_INVALID_MSG_SIZE;
-    }
 
-    if (spdm_response->header.spdm_version != spdm_request->header.spdm_version) {
-        libspdm_release_receiver_buffer (spdm_context);
-        return LIBSPDM_STATUS_INVALID_MSG_FIELD;
-    }
     if (spdm_response->header.request_response_code == SPDM_ERROR) {
         status = libspdm_handle_error_response_main(
             spdm_context, &session_id,
             &spdm_response_size, (void **)&spdm_response,
-            SPDM_KEY_UPDATE, SPDM_KEY_UPDATE_ACK,
-            sizeof(libspdm_key_update_response_mine_t));
+            SPDM_KEY_UPDATE, SPDM_KEY_UPDATE_ACK);
         if (LIBSPDM_STATUS_IS_ERROR(status)) {
             LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "SpdmVerifyKey[%x] Failed\n", session_id));
             libspdm_release_receiver_buffer (spdm_context);
             return status;
         }
     }
 
+    /* this message can only be in secured session
```

Comment:
same reason to check exact `spdm_key_update_response_t` response size after `libspdm_handle_error_response_main`